tests/history_test.go

```diff
@@ -702,14 +703,30 @@ func assertStreamConsistency(t testing.TB, clients []*redis.Client, stream strin
 	}
 }
 
-func processCheckResult(t *testing.T, client *utils.Icinga2Client, hostname string, status int) time.Time {
+// processCheckResult sends a passive check result to the Icinga2 API and returns the time of the check result.
+// The provided hostOrServiceName must be either a host name or a service name in the format "host!service".
+func processCheckResult(t *testing.T, client *utils.Icinga2Client, hostOrServiceName string, status int, isHost bool) time.Time {
 	// Ensure that check results have distinct timestamps in millisecond resolution.
 	time.Sleep(10 * time.Millisecond)
 
+	var (
+		filter  string
+		objType string
+	)
+
+	if isHost {
+		objType = "Host"
+		filter = fmt.Sprintf(`host.name==%q`, hostOrServiceName)
+	} else {
+		objType = "Service"
+		parts := strings.Split(hostOrServiceName, "!")
+		filter = fmt.Sprintf(`service.name==%q&&host.name==%q`, parts[1], parts[0])
+	}
```

Comment:
If touching that part anyways: instead of using a filter expression, it's much simpler to just add `{"host":"example.com"}` or `{"service":"example.com!ping"}` to the JSON.

Reply:
Yes, that would be enough, then you've to always make sure that you don't set the `host` and `service` JSON keys at the same time or trigger the request with one of them set to empty string. And that alone is never _simpler_ than just using a well supported filter string.

Reply:
Done!